src/daft-core/src/array/from.rs

```diff
@@ -284,47 +284,48 @@ impl TryFrom<(&str, &[Option<Series>])> for ListArray {
     }
 }
 
-impl<T: NumericNative> From<(&str, &[&[T]])> for ListArray
-where
-    T::DAFTTYPE: DaftNumericType<Native = T> + DaftArrowBackedType,
-    ArrayWrapper<DataArray<T::DAFTTYPE>>: SeriesLike,
-{
-    fn from(item: (&str, &[&[T]])) -> Self {
-        let (name, data) = item;
-        let flat_child_vec = data.iter().copied().flatten().copied().collect::<Vec<_>>();
+impl ListArray {
+    pub fn from_slice<T>(name: &str, data: &[Option<&[T]>]) -> Self
+    where
+        T: NumericNative,
+        T::DAFTTYPE: DaftNumericType<Native = T>,
+        ArrayWrapper<DataArray<T::DAFTTYPE>>: SeriesLike,
+    {
+        let flat_child_vec = data
+            .iter()
+            .copied()
+            .flatten()
+            .flatten()
+            .copied()
+            .collect::<Vec<_>>();
         let flat_child = DataArray::<T::DAFTTYPE>::from((name, flat_child_vec)).into_series();
 
-        let lengths = data.iter().map(|d| d.len());
+        let lengths = data.iter().map(|d| d.map_or(0, |d| d.len()));
         let offsets = arrow2::offset::Offsets::try_from_lengths(lengths)
             .unwrap()
             .into();
 
+        let validity =
+            arrow2::bitmap::Bitmap::from_trusted_len_iter(data.iter().map(Option::is_some));
+
         Self::new(
             flat_child.field().to_list_field().rename(name),
             flat_child,
             offsets,
-            None,
+            Some(validity),
         )
     }
-}
 
-impl<T: NumericNative> From<(&str, &[Option<&[T]>])> for ListArray
-where
-    T::DAFTTYPE: DaftNumericType<Native = T>,
-    ArrayWrapper<DataArray<T::DAFTTYPE>>: SeriesLike,
-{
-    fn from(item: (&str, &[Option<&[T]>])) -> Self {
-        let (name, data) = item;
-        let flat_child_vec = data
-            .iter()
-            .copied()
-            .flatten()
-            .flatten()
-            .copied()
-            .collect::<Vec<_>>();
+    pub fn from_vec<T>(name: &str, data: Vec<Option<Vec<T>>>) -> Self
```

Comment:
I'm curious if we could have `from_slice` take in a `&[Option<AsRef<[T]>>]` so that we don't need to implement this method. If not then no big deal.